modules/robin/src/test/ts/browser/zone/ZoneWalkerTest.ts

```diff
@@ -0,0 +1,49 @@
+import { after, before, describe, it } from '@ephox/bedrock-client';
+import { DomUniverse } from '@ephox/boss';
+import { Insert, Remove, SugarBody, SugarElement, SugarText } from '@ephox/sugar';
+import { assert } from 'chai';
+import { ZoneViewports } from 'ephox/robin/api/general/ZoneViewports';
+import * as ZoneWalker from 'ephox/robin/zone/ZoneWalker';
+
+const hook = (html: string) => {
+  let element: SugarElement<HTMLElement> | undefined;
```

Comment:
Maybe we should move the `ViewBlock` logic from TinyMCE to Agar instead, as this is largely just duplicating that?